File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/NowFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class NowFunction : NoArgumentFunction
    {
        public const string Name = "NOW";

        protected override VariableType ResultType => PrimitiveType.Date;

        private NowFunction(SourceReference reference)
            : base(reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference) => new NowFunction(reference);
    }
}

File: typescipt/src/language/expressions/functions/nowFunction.ts


export class NowFunction extends NoArgumentFunction {
   public const string Name = `NOW`;

   protected override VariableType ResultType => PrimitiveType.date;

   constructor(reference: SourceReference)
     {
     super(reference);
   }

   public static create(reference: SourceReference): ExpressionFunction {
     return new NowFunction(reference);
   }
}
